pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCacheProvider.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.config.provider;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import javax.annotation.Nullable;
+import org.apache.commons.collections4.MapUtils;
+import org.apache.pinot.common.request.Expression;
+import org.apache.pinot.spi.config.provider.LogicalTableConfigChangeListener;
+import org.apache.pinot.spi.config.provider.SchemaChangeListener;
+import org.apache.pinot.spi.config.provider.TableConfigChangeListener;
+import org.apache.pinot.spi.config.table.QueryConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.data.LogicalTableConfig;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.utils.TimestampIndexUtils;
+import org.apache.pinot.sql.parsers.CalciteSqlParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public interface TableCacheProvider {
+  Logger LOGGER = LoggerFactory.getLogger(TableCacheProvider.class);
+  boolean isIgnoreCase();
+
+  @Nullable
+  String getActualTableName(String tableName);
+
+
+  @Nullable
+  String getActualLogicalTableName(String logicalTableName);
+
+
+  Map<String, String> getTableNameMap();
+
+
+  Map<String, String> getLogicalTableNameMap();
+
+
+  List<String> getAllDimensionTables();
+
+
+  Map<String, String> getColumnNameMap(String rawTableName);
+
+
+  Map<Expression, Expression> getExpressionOverrideMap(String physicalOrLogicalTableName);
+
+
+  Set<String> getTimestampIndexColumns(String tableNameWithType);
+
+
+  List<LogicalTableConfig> getLogicalTableConfigs();
+
+
+  boolean isLogicalTable(String logicalTableName);
```

Comment:
nit: bunch of extra empty lines here, can you format this using Pinot checkstyle?